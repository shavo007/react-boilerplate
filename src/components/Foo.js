import React, { Component, Suspense, lazy } from 'react';
import { Button } from 'baseui/button';

const LoadableBar = lazy(() => import('./Bar'));

export default class Foo extends Component {
  state = { Toastify: null };

  loadToastify = () => {
    import(/* webpackChunkName: "Toastify" */ './Toastify').then(Toastify =>
      this.setState({ Toastify: Toastify.default })
    );
  };

  render() {
    const { Toastify } = this.state;

    return (
      <div>
        <h1>Showcase code splitting</h1>
        <Button
          onClick={() => {
            this.loadToastify();
          }}
        >
          Click me to demo dynamic import
        </Button>
        {Toastify ? <Toastify /> : null}

        <Suspense fallback={<div>Loading...</div>}>
          <LoadableBar />
        </Suspense>
      </div>
    );
  }
}
