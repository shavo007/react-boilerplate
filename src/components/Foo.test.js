import Foo from "./Foo";
import React from "react";

it("should match snapshot", () => {
  const wrapper = shallow(<Foo />);
  console.log(wrapper.debug());

  expect(wrapper.find("h1").text()).toBe("Showcase code splitting");
  expect(wrapper).toMatchSnapshot();
});
