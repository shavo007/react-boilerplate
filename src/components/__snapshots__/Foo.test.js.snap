// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <h1>
    Showcase code splitting
  </h1>
  <Button
    disabled={false}
    isLoading={false}
    isSelected={false}
    kind="primary"
    onClick={[Function]}
    overrides={Object {}}
    shape="default"
    size="default"
  >
    Click me to demo dynamic import
  </Button>
  <React.Suspense
    fallback={
      <div>
        Loading...
      </div>
    }
  >
    <React.Lazy />
  </React.Suspense>
</div>
`;
